ceno_zkvm/examples/fibonacci_elf.rs

```diff
@@ -0,0 +1,333 @@
+use mpcs::{Basefold, BasefoldRSParams, PolynomialCommitmentScheme};
+use std::{
+    collections::{HashMap, HashSet},
+    panic,
+    time::Instant,
+};
+use tracing_flame::FlameLayer;
+use tracing_subscriber::{EnvFilter, Registry, fmt, layer::SubscriberExt};
+use transcript::Transcript;
+
+/// Prove the execution of a fixed RISC-V program.
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Args {
+    /// The maximum number of steps to execute the program.
+    #[arg(short, long)]
+    max_steps: Option<usize>,
+}
+
+fn main() {
+    let args = Args::parse();
+
+    type E = GoldilocksExt2;
+    type Pcs = Basefold<GoldilocksExt2, BasefoldRSParams>;
+    const PROGRAM_SIZE: usize = 1 << 14;
+    type ExampleProgramTableCircuit<E> = ProgramTableCircuit<E, PROGRAM_SIZE>;
+
+    // set up logger
+    let (flame_layer, _guard) = FlameLayer::with_file("./tracing.folded").unwrap();
+    let subscriber = Registry::default()
+        .with(
+            fmt::layer()
+                .compact()
+                .with_thread_ids(false)
+                .with_thread_names(false),
+        )
+        .with(EnvFilter::from_default_env())
+        .with(flame_layer.with_threads_collapsed(true));
+    tracing::subscriber::set_global_default(subscriber).unwrap();
+
+    let sp1_platform = Platform {
+        rom_start: 0x0020_0800,
+        rom_end: 0x003f_ffff,
+        ram_start: 0x0020_0000,
+        ram_end: 0xffff_ffff,
+        unsafe_ecall_nop: true,
+    };
+    const STACK_TOP: u32 = 0x0020_0400;
+    const STACK_SIZE: u32 = 256;
+
+    let elf_bytes = include_bytes!(r"fibonacci.elf");
+    let mut vm = VMState::new_from_elf(sp1_platform, elf_bytes).unwrap();
+
+    // keygen
+    let pcs_param = Pcs::setup(1 << MAX_NUM_VARIABLES).expect("Basefold PCS setup");
+    let (pp, vp) = Pcs::trim(pcs_param, 1 << MAX_NUM_VARIABLES).expect("Basefold trim");
+    let mut zkvm_cs = ZKVMConstraintSystem::default();
+
+    let config = Rv32imConfig::<E>::construct_circuits(&mut zkvm_cs);
+    let mmu_config = MmuConfig::<E>::construct_circuits(&mut zkvm_cs);
+    let dummy_config = DummyExtraConfig::<E>::construct_circuits(&mut zkvm_cs);
+    let prog_config = zkvm_cs.register_table_circuit::<ExampleProgramTableCircuit<E>>();
+    zkvm_cs.register_global_state::<GlobalState>();
+
+    let mut zkvm_fixed_traces = ZKVMFixedTraces::default();
+
+    zkvm_fixed_traces.register_table_circuit::<ExampleProgramTableCircuit<E>>(
+        &zkvm_cs,
+        &prog_config,
+        vm.program(),
+    );
+
+    let program_data_init = {
+        let program_addrs = vm
+            .program()
+            .image
+            .iter()
+            .map(|(addr, value)| MemInitRecord {
+                addr: *addr,
+                value: *value,
+            });
+
+        let stack_addrs = (1..=STACK_SIZE)
+            .map(|i| STACK_TOP - i * WORD_SIZE as u32)
+            .map(|addr| MemInitRecord { addr, value: 0 });
```

Comment:
why does stack occurs in program data section?

Reply:
The stack is also a static allocation of the program. This program does not mention it in ELF headers though, but only as a constant `STACK_TOP` somewhere, so we have to repeat it here.

I renamed this from "program data" to just "memory", to not confuse it with the ".data" section name.

Reply:
I added a comment on this.